src/main/java/com/maarten551/tictactoe_backend/logic/LobbyContainer.java

```diff
@@ -0,0 +1,87 @@
+package com.maarten551.tictactoe_backend.logic;
+
+import com.maarten551.tictactoe_backend.exception.*;
+import com.maarten551.tictactoe_backend.model.Lobby;
+import com.maarten551.tictactoe_backend.model.Player;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+public class LobbyContainer {
+    public static final int MINIMUM_AMOUNT_OF_PLAYERS_TO_START_GAME = 2;
+    private final GameSessionContainer gameSessionContainer;
+    private ArrayList<Lobby> activeLobbies;
+
+    public LobbyContainer(GameSessionContainer gameSessionContainer) {
+        this.gameSessionContainer = gameSessionContainer;
+
+        this.activeLobbies = new ArrayList<>();
+    }
+
+    public Lobby createNewLobby(Player leader, String lobbyName) {
+        Lobby createdLobby = new Lobby(leader);
+        createdLobby.gameSession = this.gameSessionContainer.createGameSession(createdLobby);
+
+        createdLobby.name = lobbyName;
+
+        this.activeLobbies.add(createdLobby);
+
+        return createdLobby;
+    }
+
+    public List<Lobby> getAllWaitingLobbies() {
+        return this.activeLobbies.stream().filter(lobby -> !lobby.gameSession.isActive)
+                .collect(Collectors.toList());
+    }
+
+    public Optional<Lobby> getLobbyByPlayer(Player player) {
+        Optional<Lobby> lobbyByLeader = this.activeLobbies.stream().filter(lobby -> lobby.leader == player).findFirst();
+        if (lobbyByLeader.isPresent()) {
+            return lobbyByLeader;
+        }
+
+        return this.activeLobbies.stream().filter(lobby -> lobby.players.contains(player)).findFirst();
```

Comment:
What happens when no lobbies can be found  that contain player?

Reply:
Dan returned die een lege optional